core/jobs/batch_jobs/exploration_population_jobs.py

```diff
@@ -0,0 +1,312 @@
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Jobs for updating exploration model."""
+
+from __future__ import annotations
+
+import logging
+
+from core import feconf
+from core.domain import caching_services
+from core.domain import exp_domain
+from core.domain import exp_fetchers
+from core.domain import exp_services
+from core.jobs import base_jobs
+from core.jobs.io import ndb_io
+from core.jobs.transforms import job_result_transforms
+from core.jobs.types import job_run_result
+from core.platform import models
+
+import apache_beam as beam
+import result
+
+from typing import Iterable, Sequence, Tuple
+
+MYPY = False
+if MYPY: # pragma: no cover
+    from mypy_imports import base_models
+    from mypy_imports import datastore_services
+    from mypy_imports import exp_models
+
+(base_models, exp_models,) = models.Registry.import_models([
+    models.NAMES.base_model, models.NAMES.exploration
+])
+datastore_services = models.Registry.import_datastore_services()
+
+
+class PopulateExplorationWithAndroidProtoSizeInBytesJob(base_jobs.JobBase):
+    """Job that populate exploration models with
+    android_proto_size_in_bytes attribute.
+    """
+
+    @staticmethod
+    def _migrate_exploration(
+        exp_id: str, exp_model: exp_models.ExplorationModel
+    ) -> result.Result[
+        Tuple[str, exp_domain.Exploration], Tuple[str, Exception]
+    ]:
+        """Migrates exploration and transforms exploration model
+        into exploration object.
+
+        Args:
+            exp_id: str. The id of the exploration.
+            exp_model: ExplorationModel. The exploration model to migrate.
+
+        Returns:
+            Result((str, Exploration), (str, Exception)). Result
+            containing tuple that consists of exploration ID and either
+            exploration object or Exception. Exploration object is
+            returned when the migration was successful and Exception
+            is returned otherwise.
+        """
+        try:
+            exploration = exp_fetchers.get_exploration_from_model(exp_model)
+            exploration.validate()
+        except Exception as e:
+            logging.exception(e)
+            return result.Err((exp_id, e))
+
+        return result.Ok((exp_id, exploration))
+
+    @staticmethod
+    def _update_exploration(
+        exp_model: exp_models.ExplorationModel,
+        migrated_exploration: exp_domain.Exploration,
+        exploration_changes: Sequence[exp_domain.ExplorationChange],
+        exp_rights_model: exp_models.ExplorationRightsModel
+    ) -> Sequence[base_models.BaseModel]:
+        """Generates newly updated exploration models.
+
+        Args:
+            exp_model: ExplorationModel. The exploration which
```

Comment:
"exploration" usually refers to the domain object. If you're talking about the exploration model, say "exploration model".

Reply:
Fixed, this is about the model.